src/test/java/io/odpf/firehose/sink/prometheus/builder/WriteRequestBuilderTest.java

```diff
@@ -0,0 +1,110 @@
+        sample1 = Cortex.Sample.newBuilder().setTimestampMs(10000).setValue(100).build();
+        sample2 = Cortex.Sample.newBuilder().setTimestampMs(8000).setValue(200).build();
+    }
+
+    @Test
+    public void shouldReturnWriteRequest() throws InvalidProtocolBufferException {
+        Cortex.TimeSeries timeSeries = Cortex.TimeSeries.newBuilder()
+                .addLabels(labelPair1).addSamples(sample1)
+                .addLabels(labelPair2)
+                .build();
+
+        Mockito.when(timeSeriesBuilder.buildTimeSeries(dynamicMessage1, message1.getPartition())).thenReturn(Collections.singletonList(timeSeries));
+        Mockito.when(protoParser.parse(message1.getLogMessage())).thenReturn(dynamicMessage1);
+
+        Cortex.WriteRequest writeRequest = new WriteRequestBuilder(timeSeriesBuilder, protoParser)
+                .buildWriteRequest(Collections.singletonList(message1));
+
+        String expected = "timeseries {\n  labels {\n    name: \"__name__\"\n    value: \"tip_amount\"\n  }\n  labels {\n    name: \"kafka_partition\"\n    value: \"0\"\n  }\n  samples {\n    value: 100.0\n    timestamp_ms: 10000\n  }\n}\n";
+        assertEquals(expected, writeRequest.toString());
+    }
+
+    @Test
+    public void shouldReturnWriteRequestWithSortedTimeSeries() throws InvalidProtocolBufferException {
+        Cortex.TimeSeries timeSeries1 = Cortex.TimeSeries.newBuilder()
+                .addLabels(labelPair1).addSamples(sample1)
+                .addLabels(labelPair2)
+                .build();
+        Cortex.TimeSeries timeSeries2 = Cortex.TimeSeries.newBuilder()
+                .addLabels(labelPair1).addSamples(sample2)
+                .addLabels(labelPair3)
+                .build();
+
+        Mockito.when(timeSeriesBuilder.buildTimeSeries(dynamicMessage1, message1.getPartition())).thenReturn(Collections.singletonList(timeSeries1));
+        Mockito.when(protoParser.parse(message1.getLogMessage())).thenReturn(dynamicMessage1);
+        Mockito.when(timeSeriesBuilder.buildTimeSeries(dynamicMessage2, message2.getPartition())).thenReturn(Collections.singletonList(timeSeries2));
+        Mockito.when(protoParser.parse(message2.getLogMessage())).thenReturn(dynamicMessage2);
+
+        Cortex.WriteRequest writeRequest = new WriteRequestBuilder(timeSeriesBuilder, protoParser)
+                .buildWriteRequest(messages);
+
+        String expected = "timeseries {\n  labels {\n    name: \"__name__\"\n    value: \"tip_amount\"\n  }\n  labels {\n    name: \"kafka_partition\"\n    value: \"1\"\n  }\n  samples {\n    value: 200.0\n    timestamp_ms: 8000\n  }\n}\n"
+                + "timeseries {\n  labels {\n    name: \"__name__\"\n    value: \"tip_amount\"\n  }\n  labels {\n    name: \"kafka_partition\"\n    value: \"0\"\n  }\n  samples {\n    value: 100.0\n    timestamp_ms: 10000\n  }\n}\n";
+        assertEquals(expected, writeRequest.toString());
+    }
+}
```

Comment:
following scenarios not covered:
1. Empty list of messages
2. Parsing error (invalid message)